lib/Runtime/Base/FunctionBody.cpp

```diff
@@ -8020,56 +8106,38 @@ namespace Js
     }
 #endif
 
-    void EntryPointInfo::JitTransferData::AddJitTimeTypeRef(void* typeRef, Recycler* recycler)
-    {
-        Assert(typeRef != nullptr);
-        EnsureJitTimeTypeRefs(recycler);
-        this->jitTimeTypeRefs->AddNew(typeRef);
-    }
-
-    void EntryPointInfo::JitTransferData::EnsureJitTimeTypeRefs(Recycler* recycler)
-    {
-        if (this->jitTimeTypeRefs == nullptr)
-        {
-            this->jitTimeTypeRefs = RecyclerNew(recycler, TypeRefSet, recycler);
-        }
-    }
-
+   
     void EntryPointInfo::PinTypeRefs(ScriptContext* scriptContext)
     {
-        Assert(this->jitTransferData != nullptr && this->jitTransferData->GetIsReady());
+        NativeEntryPointData * nativeEntryPointData = this->GetNativeEntryPointData();
+        JitTransferData * jitTransferData = nativeEntryPointData->GetJitTransferData();
+        Assert(jitTransferData != nullptr && jitTransferData->GetIsReady());
 
         Recycler* recycler = scriptContext->GetRecycler();
-        if (this->jitTransferData->GetRuntimeTypeRefs() != nullptr)
+        if (jitTransferData->GetRuntimeTypeRefs() != nullptr)
         {
             // Copy pinned types from a heap allocated array created on the background thread
             // to a recycler allocated array which will live as long as this EntryPointInfo.
             // The original heap allocated array will be freed at the end of NativeCodeGenerator::CheckCodeGenDone
-            void** jitPinnedTypeRefs = this->jitTransferData->GetRuntimeTypeRefs();
-            size_t jitPinnedTypeRefCount = this->jitTransferData->GetRuntimeTypeRefCount();
-            this->runtimeTypeRefs = RecyclerNewArray(recycler, Field(void*), jitPinnedTypeRefCount + 1);
-            //js_memcpy_s(this->runtimeTypeRefs, jitPinnedTypeRefCount * sizeof(void*), jitPinnedTypeRefs, jitPinnedTypeRefCount * sizeof(void*));
-            for (size_t i = 0; i < jitPinnedTypeRefCount; i++)
-            {
-                this->runtimeTypeRefs[i] = jitPinnedTypeRefs[i];
-            }
-            this->runtimeTypeRefs[jitPinnedTypeRefCount] = nullptr;
+            void** jitPinnedTypeRefs = jitTransferData->GetRuntimeTypeRefs();
+            size_t jitPinnedTypeRefCount = jitTransferData->GetRuntimeTypeRefCount();
+            nativeEntryPointData->PinTypeRefs(recycler, jitPinnedTypeRefCount, jitPinnedTypeRefs);
         }
     }
 
     void EntryPointInfo::InstallGuards(ScriptContext* scriptContext)
     {
-        Assert(this->jitTransferData != nullptr && this->jitTransferData->GetIsReady());
-        Assert(this->equivalentTypeCacheCount == 0 && this->equivalentTypeCaches == nullptr);
```

Comment:
Worth introducing an assert in NativeEntryPointData, something like nativeEntryPointData->AssertNoEquivalentTypeCaches()?

Reply:
These check are moved to EnsureEquivalentTypeCache and EnsurePropertyGuardWeakRefs already.